lightning-invoice/src/utils.rs

```diff
@@ -8,19 +8,115 @@ use bitcoin_hashes::Hash;
 use crate::prelude::*;
 use lightning::chain;
 use lightning::chain::chaininterface::{BroadcasterInterface, FeeEstimator};
-use lightning::chain::keysinterface::{Sign, KeysInterface};
+use lightning::chain::keysinterface::{Invoice as InvoiceType, KeysInterface, Sign};
 use lightning::ln::{PaymentHash, PaymentPreimage, PaymentSecret};
-use lightning::ln::channelmanager::{ChannelDetails, ChannelManager, PaymentId, PaymentSendFailure, MIN_FINAL_CLTV_EXPIRY};
+use lightning::ln::channelmanager::{ChannelDetails, ChannelManager, PaymentId, PaymentSendFailure, MIN_FINAL_CLTV_EXPIRY, MIN_CLTV_EXPIRY_DELTA};
 use lightning::ln::msgs::LightningError;
 use lightning::routing::scoring::Score;
 use lightning::routing::network_graph::{NetworkGraph, RoutingFees};
 use lightning::routing::router::{Route, RouteHint, RouteHintHop, RouteParameters, find_route};
 use lightning::util::logger::Logger;
+use secp256k1::Secp256k1;
 use secp256k1::key::PublicKey;
 use core::convert::TryInto;
 use core::ops::Deref;
 use core::time::Duration;
 
+#[cfg(feature = "std")]
+/// Utility to create an invoice that can be paid to one of multiple nodes, or a "phantom invoice."
+/// This works because the invoice officially pays to a fake node (the "phantom"). Useful for
+/// load-balancing payments between multiple nodes.
+///
+/// `channels` parameter:
+/// * Contains a list of tuples of `(phantom_node_scid, real_node_pubkey,
+///   real_channel_details)`, where `phantom_node_scid` is unique to each `channels` entry
+/// * `phantom_node_scid`s are retrieved from [`ChannelManager::get_phantom_scid`]
+/// * `real_channel_details` are retrieved from [`ChannelManager::list_channels`]
+/// * Contains channel info for all nodes participating in the phantom invoice
```

Comment:
Per this feedback: https://github.com/lightningdevkit/rust-lightning/pull/1199#discussion_r788972878 this was just meant to clarify that the list of tuples is sourced from multiple nodes. Not sure I parsed your interpretation though, lmk how this could be clarified

Reply:
Right, sorry, ignore this comment, somehow I'd read the list to be `(publicKey, u64, Vec<ChannelDetails>)`. That said, *should* it instead be that? Really you have a list of channels per node. In theory you can have a different phantom SCID per channel entry, but there's not much reason to, you can just have one phantom SCID per node.

Reply:
Sure, I take it that would be supported by the bindings? Will switch.

Reply:
Uhmmmmmmm, I mean, I think we can if either we do a `Vec<ChannelDetails>`  or a `&[&ChannelDetails]` just not `&[ChannelDetails]`. I dont have a strong preference, in the past we've done the second, but I believe we've had a discussion about moving to the first cause it cleans up callsites a lot, even if it can cause some clones on the user's end. cc @jkczyz

Reply:
IIRC, we wanted `&Vec<T>` or maybe `&[T]` to simplify passing an event's `Vec<T>` field. With `&[&T]` you need to first do `.iter().collect::<Vec<_>>()`.